File: app/src/test/java/com/hundaol/ethiopic/cal/GregorianCalTest_20170513.java
package com.hundaol.ethiopic.cal;

import org.junit.Test;

import static org.hamcrest.CoreMatchers.is;
import static org.junit.Assert.assertThat;

/**
 * Created by john.pirie on 2017-04-22.
 */
public class GregorianCalTest_20170513 {

    private ICal cal = GregorianCal.INSTANCE;
    private int jdn = GregorianCal.INSTANCE.fromDate(2017, 5, 13);

    @Test
    public void fromDate() {
        assertThat(jdn, is(2457887));
    }

    // date

    @Test
    public void getYear() {
        assertThat(cal.getYear(jdn), is(2017));
    }

    @Test
    public void getMonth() {
        assertThat(cal.getMonth(jdn), is(5));
    }

    @Test
    public void getDay() {
        assertThat(cal.getDay(jdn), is(13));
    }

    // day

    @Test
    public void nextDay() {
        assertThat(cal.nextDay(jdn, 20), is(cal.fromDate(2017, 6, 2)));
    }

    @Test
    public void prevDay() {
        assertThat(cal.prevDay(jdn, 20), is(cal.fromDate(2017, 4, 23)));
    }

    // week

    @Test
    public void getDayOfWeek() {
        assertThat(cal.getDayOfWeek(jdn), is(6));
    }

    @Test
    public void getFirstDayOfWeek() {
        assertThat(cal.getFirstDayOfWeek(jdn), is(cal.fromDate(2017, 5, 7)));
    }

    @Test
    public void getLastDayOfWeek() {
        assertThat(cal.getLastDayOfWeek(jdn), is(cal.fromDate(2017, 5, 13)));
    }

    @Test
    public void isWeekday() {
        assertThat(cal.isWeekday(jdn), is(false));
    }

    @Test
    public void isWeekend() {
        assertThat(cal.isWeekend(jdn), is(true));
    }

    // month

    @Test
    public void firstOfMonth() {
        assertThat(cal.firstOfMonth(jdn), is(cal.fromDate(2017, 5, 1)));
    }

    @Test
    public void lastOfMonth() {
        assertThat(cal.lastOfMonth(jdn), is(cal.fromDate(2017, 5, 31)));
    }

    @Test
    public void nextMonth() {
        assertThat(cal.nextMonth(jdn), is(cal.fromDate(2017, 6, 1)));
    }

    @Test
    public void prevMonth() {
        assertThat(cal.prevMonth(jdn), is(cal.fromDate(2017, 4, 1)));
    }

    @Test
    public void getMonthName() {
        assertThat(cal.getMonthName(jdn)[0], is('M'));
        assertThat(cal.getMonthName(jdn)[1], is('A'));
        assertThat(cal.getMonthName(jdn)[2], is('Y'));
    }

    @Test
    public void getDaysInMonth() {
        assertThat(cal.getDaysInMonth(jdn), is(31));
    }

    @Test
    public void getWeeksInMonth() {
        assertThat(cal.getWeeksInMonth(jdn), is(5));
    }

    @Test
    public void getWeekOfMonth() {
        assertThat(cal.getWeekOfMonth(jdn), is(1));
    }

    @Test
    public void getDayOfWeekOfFirstDayInMonth() {
        assertThat(cal.getDayOfWeekOfFirstDayInMonth(jdn), is(1));
    }

    @Test
    public void getLastWeekDayOfMonth() {
        assertThat(cal.getDayOfWeekOfLastDayInMonth(jdn), is(3));
    }

    @Test
    public void getLastFullWeek() {
        assertThat(cal.getLastFullWeek(jdn), is(3));
    }


    // year

    @Test
    public void getLastFullWgeteek() {
        assertThat(cal.getMonthsInYear(), is(12));
    }

    // week number

    @Test
    public void getWeekNumber() {
        assertThat(cal.getWeekNumber(jdn), is(351126));
    }
}

File: app/src/test/java/com/hundaol/ethiopic/cal/GregorianCalTest_20170513.kt
package com.hundaol.ethiopic.cal

import org.junit.Test

import org.hamcrest.CoreMatchers.`is` as Is
import org.junit.Assert.assertThat

/**
 * Created by john.pirie on 2017-04-22.
 */
class GregorianCalTest_20170513 {

    private val cal = GregorianCal.INSTANCE
    private val jdn = 2457887 // G: 2017, 5, 13 E: 2009, 9, 5

    @Test
    fun fromDate() {
        assertThat(cal.fromDate(2017, 5, 13), Is(jdn))
    }

    // date

    @Test
    fun getYear() {
        assertThat(cal.getYear(jdn), Is(2017))
    }

    @Test
    fun getMonth() {
        assertThat(cal.getMonth(jdn), Is(5))
    }

    @Test
    fun getDay() {
        assertThat(cal.getDay(jdn), Is(13))
    }

    // day

    @Test
    fun nextDay() {
        assertThat(cal.nextDay(jdn, 20), Is(cal.fromDate(2017, 6, 2)))
    }

    @Test
    fun prevDay() {
        assertThat(cal.prevDay(jdn, 20), Is(cal.fromDate(2017, 4, 23)))
    }

    // week

    @Test
    fun getDayOfWeek() {
        assertThat(cal.getDayOfWeek(jdn), Is(6))
    }

    @Test
    fun getFirstDayOfWeek() {
        assertThat(cal.getFirstDayOfWeek(jdn), Is(cal.fromDate(2017, 5, 7)))
    }

    @Test
    fun getLastDayOfWeek() {
        assertThat(cal.getLastDayOfWeek(jdn), Is(cal.fromDate(2017, 5, 13)))
    }

    @Test
    fun isWeekday() {
        assertThat(cal.isWeekday(jdn), Is(false))
    }

    @Test
    fun isWeekend() {
        assertThat(cal.isWeekend(jdn), Is(true))
    }

    // month

    @Test
    fun firstOfMonth() {
        assertThat(cal.firstOfMonth(jdn), Is(cal.fromDate(2017, 5, 1)))
    }

    @Test
    fun lastOfMonth() {
        assertThat(cal.lastOfMonth(jdn), Is(cal.fromDate(2017, 5, 31)))
    }

    @Test
    fun nextMonth() {
        assertThat(cal.nextMonth(jdn), Is(cal.fromDate(2017, 6, 1)))
    }

    @Test
    fun prevMonth() {
        assertThat(cal.prevMonth(jdn), Is(cal.fromDate(2017, 4, 1)))
    }

    @Test
    fun getMonthName() {
        assertThat(cal.getMonthName(jdn)[0], Is('M'))
        assertThat(cal.getMonthName(jdn)[1], Is('a'))
        assertThat(cal.getMonthName(jdn)[2], Is('y'))
    }

    @Test
    fun getDaysInMonth() {
        assertThat(cal.getDaysInMonth(jdn), Is(31))
    }

    @Test
    fun getWeeksInMonth() {
        assertThat(cal.getWeeksInMonth(jdn), Is(5))
    }

    @Test
    fun getWeekOfMonth() {
        assertThat(cal.getWeekOfMonth(jdn), Is(1))
    }

    @Test
    fun getDayOfWeekOfFirstDayInMonth() {
        assertThat(cal.getDayOfWeekOfFirstDayInMonth(jdn), Is(1))
    }

    @Test
    fun getLastWeekDayOfMonth() {
        assertThat(cal.getDayOfWeekOfLastDayInMonth(jdn), Is(3))
    }

    @Test
    fun getLastFullWeek() {
        assertThat(cal.getLastFullWeek(jdn), Is(3))
    }


    // year

    @Test
    fun getLastFullWgeteek() {
        assertThat(cal.monthsInYear, Is(12))
    }

    // week number

    @Test
    fun getWeekNumber() {
        assertThat(cal.getWeekNumber(jdn), Is(351126))
    }
}
